test/integration/good/code-gen/cl.expected

```diff
@@ -1392,6 +1392,24 @@ class optimize_glm_model final : public model_base_crtp<optimize_glm_model> {
     // suppress unused var warning
     (void) DUMMY_VAR__;
     try {
+      context__.validate_dims("parameter initialization", "alpha_v",
+        "double", std::vector<size_t>{static_cast<size_t>(k)});
```

Comment:
I don't see assignment to `current_statement__` so if this fails the error is going to say something like "(found before start of program)" instead of a line number, right? Would it be possible to include location of the declaration in Program.outvar and add `Numbering.assign_loc` right before the `validate_dims` call?

Reply:
Yeah, currently transform inits basically lacks any use of the locating mechanism. E.g, try:

```stan
parameters {
  simplex[2] s;
}
```
```json
{ "s": [ 0.3, 0.8 ] }
```

The output says 
```
Exception: stan::math::simplex_free: Simplex variable is not a valid simplex. sum(Simplex variable) = 1.1, but should be 1 (found before start of program)
```

Reply:
I was able to make this change but it was kind of big/annoying. Starting a separate branch if that works for you?